### PR TITLE
add TSRM header in zts mode and module pointer

### PR DESCRIPTION
**Pull request by krakjoe**

### CODE REVIEW

File: lib/ReckiCT/Compiler/PECL/Compiler.php [hunk 1]
```diff
@@ -98,6 +98,10 @@ protected function generateHeaderFile($moduleName) {
         $code .= "#define PHP_{$upperName}_VERSION \"1.0\"\n";
         $code .= "#define PHP_{$upperName}_EXTNAME \"{$moduleName}\"\n";
         $code .= "extern zend_module_entry {$moduleName}_module_entry;\n";
+        $code .= "#define phpext_{$moduleName}_ptr &{$moduleName}_module_entry\n";
```
Comment (Ocramius) [inline lib/ReckiCT/Compiler/PECL/Compiler.php:101]: May want to add some comments here, explaining the reason for these define/include blocks...

**Comment by Ocramius** [comment]
The code in the diff seems to be uncovered by a unit test...

File: lib/ReckiCT/Compiler/PECL/Compiler.php (hunk 1 above)
Comment (krakjoe) [inline lib/ReckiCT/Compiler/PECL/Compiler.php:101]: That's included in skeletons and used by deps, and possibly windows ...

Comment (krakjoe) [inline lib/ReckiCT/Compiler/PECL/Compiler.php:101]: The TSRM.h header should be included before anything else in ZTS if we don't want weirdness, again just best practice ...

Comment (Ocramius) [inline lib/ReckiCT/Compiler/PECL/Compiler.php:101]: Take that comment and make it a code comment :-)

**Comment by bwoebi** [comment]
@Ocramius the whole branch is only experimental... Tests are for later.

**Pull request merged**